### PR TITLE
Add UG and DG docs for Reschedule Command

### PR DESCRIPTION
**Pull request by Dethada**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -226,6 +226,25 @@ Scheduled meeting with John Doe from 23-02-2024 16:00 to 23-02-2024 17:00
...
+Format: `reschedule INDEX [s/START_DATETIME] [e/END_DATETIME] [mr/MEETING_REMARK]`
+
+* The index refers to the index number shown in the displayed person list.
+* The index **must be a positive integer** 1, 2, 3, …​
+* Specified person must have a meeting scheduled.
+* At least one of the optional fields must be provided.
+* Input values will overwrite existing value that were assigned to the specified field.
+* `START_DATETIME` must be after the system's current datetime.
+* `END_DATETIME` must be after `START_DATETIME`
+* `MEETING_REMARK` has the same constraints as `REMARK` as stated above.
```
Comment (KohGuanZeh) [inline docs/UserGuide.md:242]: I think REMARK does not have a length limit like MEETING_REMARK? I think we should include the character limit or add an additional MEETING_REMARK constraints above that includes the length character limit and allowed set of characters?

Comment (Dethada) [reply docs/UserGuide.md]: Added MEETING_REMARK format to the top of UG

**Review by Javiery3889 — APPROVED** [review]

**Review by KohGuanZeh — APPROVED** [review]

File: docs/UserGuide.md [hunk 2]
```diff
@@ -332,6 +350,7 @@ Action | Format, Examples
 **Find** | `find n/NAME|e/EMAIL|p/PHONE_NUMBER|a/ADDRESS|r/REMARK|m/MEETING_DATE`<br>`|mr/MEETING_REMARK|t/TAG…`<br> e.g., `find n/Alice Tan`
 **List** | `list`
 **Schedule** | `schedule INDEX s/START_DATETIME e/END_DATETIME [mr/MEETING_REMARK]`<br> e.g., `schedule 1 s/23-02-2024T16:00 e/23-02-2024T17:00 mr/online meeting`
+**Reschedule** | `reschedule INDEX [s/START_DATETIME] [e/END_DATETIME] [mr/MEETING_REMARK]`<br> e.g., `reschedule 1 s/23-02-2024T16:00 e/23-02-2024T17:00`
```
Comment (YuZizhen) [inline docs/UserGuide.md:353]: Is it better to omit the brackets [] as the rest all don't have

Comment (KohGuanZeh) [reply docs/UserGuide.md:353]: The brackets are required since its following the conventions. I think this is fine.

**Review by Javiery3889 — APPROVED** [review]
LGTM

**Pull request merged**